tools/bridge/bridge/config.py

```diff
@@ -110,47 +110,44 @@ def validate_host_and_port_if_enabled(self, in_data, **kwargs):
                 )
 
 
-class ConfigSchema(Schema):
-    home_rpc_url = fields.Url(required=True)
-    home_bridge_contract_address = AddressField(required=True)
-    foreign_rpc_url = fields.Url(required=True)
-    foreign_chain_token_contract_address = AddressField(required=True)
-    foreign_bridge_contract_address = AddressField(required=True)
-    validator_private_key = fields.Nested(PrivateKeySchema, required=True)
-
-    logging = LoggingField(missing=lambda: dict(FORCED_LOGGING_CONFIG))
-    home_rpc_timeout = fields.Integer(missing=180, validate=validate_non_negative)
-    home_chain_gas_price = fields.Integer(
-        missing=10 * 1000000000, validate=validate_non_negative
-    )  # Gas price is in GWei
-    home_chain_max_reorg_depth = fields.Integer(
-        missing=1, validate=validate_non_negative
-    )
-    home_chain_event_poll_interval = fields.Float(
-        missing=5, validate=validate_non_negative
-    )
-    home_chain_event_fetch_start_block_number = fields.Integer(
+class ChainSchema(Schema):
+    rpc_url = fields.Url(required=True)
+    rpc_timeout = fields.Integer(missing=180, validate=validate_non_negative)
+    bridge_contract_address = AddressField(required=True)
+    max_reorg_depth = fields.Integer(validate=validate_non_negative)
+    event_poll_interval = fields.Float(missing=5, validate=validate_non_negative)
+    event_fetch_start_block_number = fields.Integer(
         missing=0, validate=validate_non_negative
     )
-    foreign_rpc_timeout = fields.Integer(missing=180, validate=validate_non_negative)
-    foreign_chain_max_reorg_depth = fields.Integer(
-        missing=10, validate=validate_non_negative
-    )
-    foreign_chain_event_poll_interval = fields.Integer(
-        missing=5, validate=validate_non_negative
-    )
-    foreign_chain_event_fetch_start_block_number = fields.Integer(
-        missing=0, validate=validate_non_negative
-    )
-    balance_warn_poll_interval = fields.Float(
-        missing=60, validate=validate_non_negative
+
+
+class ForeignChainSchema(ChainSchema):
+    max_reorg_depth = fields.Integer(missing=1, validate=validate_non_negative)
+    token_contract_address = AddressField(required=True)
+
+
+class HomeChainSchema(ChainSchema):
+    max_reorg_depth = fields.Integer(missing=10, validate=validate_non_negative)
+    gas_price = fields.Integer(
+        missing=to_wei(10, "ether"), validate=validate_non_negative
```

Comment:
10 ether gas price?

Reply:
:scream: fixed

Reply:
Should we check that the gas price is within reasonable bounds? In order to prevent the user from configuring something like we had here?